internal/internal_workflow_client.go

```diff
@@ -483,41 +484,62 @@ func (wc *workflowClient) TerminateWorkflow(ctx context.Context, workflowID stri
 // GetWorkflowHistory return a channel which contains the history events of a given workflow
 func (wc *workflowClient) GetWorkflowHistory(ctx context.Context, workflowID string, runID string,
 	isLongPoll bool, filterType s.HistoryEventFilterType) HistoryEventIterator {
+	if isLongPoll {
+		return wc.PollWorkflowHistory(ctx, workflowID, runID, filterType)
+	} else {
+		return wc.GetWorkflowRawHistory(ctx, workflowID, runID)
+	}
+}
 
+// PollWorkflowHistory performs long polling of the history blob data from server and deserialize to history event construct data
+// workflowID is required, other parameters are optional.
+// If runID is omit, it will terminate currently running workflow (if there is one) based on the workflowID.
+func (wc *workflowClient) PollWorkflowHistory(ctx context.Context, workflowID string, runID string,
+	filterType s.HistoryEventFilterType) HistoryEventIterator {
 	domain := wc.domain
 	paginate := func(nexttoken []byte) (*s.GetWorkflowExecutionHistoryResponse, error) {
-		request := &s.GetWorkflowExecutionHistoryRequest{
+
+		var response *s.GetWorkflowExecutionHistoryResponse
+		var err error
+		request := &s.PollForWorkflowExecutionRawHistoryRequest{
 			Domain: common.StringPtr(domain),
 			Execution: &s.WorkflowExecution{
 				WorkflowId: common.StringPtr(workflowID),
 				RunId:      getRunID(runID),
 			},
-			WaitForNewEvent:        common.BoolPtr(isLongPoll),
 			HistoryEventFilterType: &filterType,
 			NextPageToken:          nexttoken,
 		}
-
-		var response *s.GetWorkflowExecutionHistoryResponse
-		var err error
 	Loop:
 		for {
 			err = backoff.Retry(ctx,
 				func() error {
```

Comment:
this func() is simply too long for it to be an anonymous inline func. Can you create a separate helper method and pass that method to backoff.Retry() ?